src/v/json/_include_first.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2022 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "likely.h"
+
+#include <exception>
+#include <iostream>
+
+#define RAPIDJSON_HAS_STDSTRING 1
+#define RAPIDJSON_ASSERT(x)                                                    \
+    do {                                                                       \
+        if (unlikely(!(x))) {                                                  \
+            std::cerr << "Rapidjson failure: " << __FILE__ << ":" << __LINE__  \
```

Comment:
Just moving code, but yeah, I think vassert would be nicer, happy to change it in this PR

Reply:
I don't know macros too well, but something like like:
```cpp
RAPIDJSON_ASSERT(x) vassert(x, "rapidjson: ")
```
Should be safe, since they use the same do..while trick, right?

In which case I'm almost tempted to add the defines to cmake and do away with the awkward `_include_first.h` file.

Reply:
Correct. I think that will work.

Reply:
You can also add the defined on cmake at the rapid json import level.

Reply:
That way is not global and whoever uses it will get the definition imported transparently

Reply:
Yes, that was the idea, I think a target INTERFACE property would do it.

Reply:
kinda feel like we should avoid injecting code/statements through the command line, but I don't have any good reason for holding that opinion. did forcing all the json imports to go through a header we control with the macro set first not cover all the cases?

Reply:
Yeah, anything that `vassert` depends on won't be included. With most of the removal of rapidjson includes, the macro redefinition problem is much reduced. I'll just switch it to vassert, then.

Reply:
Can you please explain to me this trick with `do..while`? Why do we need it?

Reply:
It's to make the compound statement a single, so that you can do:
```c++
if(x == y)
    RAPIDJSON_ASSERT(x == y ));
else
    // ...
```
And the second statement isn't "unnested" and run unconditionally:

We always but the scope on, but mistakes happen and this is a widely accepted pattern for compound statement macros. Reason 432 not to use macros.